components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractLocalApplicationAuthenticator.java

```diff
@@ -0,0 +1,289 @@
+import org.wso2.carbon.identity.application.authentication.framework.internal.FrameworkServiceDataHolder;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.core.model.IdentityErrorMsgContext;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.event.IdentityEventConstants;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.event.services.IdentityEventService;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class AbstractLocalApplicationAuthenticator extends AbstractApplicationAuthenticator
+        implements ApplicationAuthenticator {
+
+    private static final long serialVersionUID = -4406878411547612129L;
+    private static final Log log = LogFactory.getLog(AbstractLocalApplicationAuthenticator.class);
+
+    @Override
+    public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletResponse response,
+                                           AuthenticationContext context) throws AuthenticationFailedException,
+            LogoutFailedException {
+
+        // if an authentication flow
+        if (!context.isLogoutRequest()) {
+            if (!canHandle(request)
+                    || Boolean.TRUE.equals(request.getAttribute(FrameworkConstants.REQ_ATTR_HANDLED))) {
+                context.setRetrying(false);
+                return initiateAuthenticationFlow(request, response, context);
+            } else {
+                try {
+                    fireEvent(context, IdentityEventConstants.Event.PRE_AUTHENTICATION, false);
+                    processAuthenticationResponse(request, response, context);
+                    if (this instanceof LocalApplicationAuthenticator && !context.getSequenceConfig()
+                            .getApplicationConfig().isSaaSApp()) {
+                        processWithNonSaaSApp(context);
+                    }
+                    request.setAttribute(FrameworkConstants.REQ_ATTR_HANDLED, true);
+                    context.setProperty(FrameworkConstants.LAST_FAILED_AUTHENTICATOR, null);
+                    fireEvent(context, IdentityEventConstants.Event.POST_AUTHENTICATION, true);
+                    return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+                } catch (AuthenticationFailedException e) {
+                    if (checkUserAccountStatus(response, context)) {
```

Comment:
According to this, If the account is locked/disabled, we are not firing the post login event?